exercises/largest-series-product/test/test_largest_series_product.c

```diff
@@ -0,0 +1,114 @@
+#include "vendor/unity.h"
+#include "../src/largest_series_product.h"
+
+void test_can_find_the_largest_product_of_2_with_numbers_in_order(void)
+{
+   TEST_ASSERT_EQUAL(72, largest_series_product("0123456789", 2));
+}
+
+void test_can_find_the_largest_product_of_2(void)
+{
+   TEST_ASSERT_EQUAL(48, largest_series_product("576802143", 2));
+}
+
+void test_finds_the_largest_product_if_span_equals_length(void)
+{
+   TEST_ASSERT_EQUAL(18, largest_series_product("29", 2));
+}
+
+void test_can_find_the_largest_product_of_3_with_numbers_in_order(void)
+{
+   TEST_ASSERT_EQUAL(504, largest_series_product("0123456789", 3));
+}
+
+void test_can_find_the_largest_product_of_3(void)
+{
+   TEST_ASSERT_EQUAL(270, largest_series_product("1027839564", 3));
+}
+
+void test_can_find_the_largest_product_of_5_with_numbers_in_order(void)
+{
+   TEST_ASSERT_EQUAL(15120, largest_series_product("0123456789", 5));
+}
+
+void test_can_get_the_largest_product_of_a_big_number(void)
+{
+   TEST_ASSERT_EQUAL(23520,
+                     largest_series_product
+                     ("73167176531330624919225119674426574742355349194934", 6));
+}
+
+void test_can_get_the_largest_product_of_a_big_number_ii(void)
+{
+   TEST_ASSERT_EQUAL(28350,
+                     largest_series_product
+                     ("52677741234314237566414902593461595376319419139427", 6));
+}
+
+void test_can_get_the_largest_product_of_a_big_number_project_euler(void)
+{
+   TEST_ASSERT_EQUAL(23514624000,
```

Comment:
`TEST_ASSERT_EQUAL` is a macro define for `TEST_ASSERT_EQUAL_INT`, which may overflow here. It would still pass the tests since it just casts them both to an int. (I think it uses 64bit ints by default on systems that it detects have 64bit pointers, but I think it's safer to explicitly use 64bit types where we know we need it.)